### PR TITLE
feat: save as ium

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**
Ref: https://github.com/zakodium-oss/analysis-ui-components/issues/216

### CODE REVIEW

**Comment by lpatiny** [comment]
Functionality looks ok to me.

File: src/app/context/appState.tsx [hunk 1]
```diff
@@ -157,7 +162,23 @@ function actionHandler(draft: Draft<AppState>, action: AppStateAction) {
       draft.isLoading = false;
       return;
     }
+    case 'SAVE_AS_IUM': {
+      saveAsIum(draft, action.payload?.filename, action.payload?.spaceIndent);
+      return;
+    }
```
Comment (stropitek) [inline src/app/context/appState.tsx:168]: You should not have side effects in a reducer.

Actions should only update the state.

File: src/app/App.tsx [hunk 2]
```diff
@@ -69,6 +70,9 @@ function DropZoneArea() {
     { id: '1h,1h', title: '1H,1H', content: 'Hello, World! [c]' },
     { id: '1h,13c', title: '1H,13C', content: 'Hello, World! [d]' },
   ];
+  function saveHandler() {
+    dispatch({ type: 'SAVE_AS_IUM' });
+  }
```
Comment (stropitek) [inline src/app/App.tsx:75]: Please implement the download logic here, not in the reducer. Please implement the download without the `file-saver` library.

It's relatively easy to do, something like this:

```ts
export function download(data: string) {
  const link = document.createElement('a');

  const blob = new Blob([data]);
  const url = URL.createObjectURL(blob);


  link.setAttribute('href', url);
  link.setAttribute('download', name);
  link.style.visibility = 'hidden';
  document.body.appendChild(link);
  link.click();
  document.body.removeChild(link);
  URL.revokeObjectURL(url);
}
```

Comment (wadjih-bencheikh18) [reply src/app/App.tsx]: fixed thank you

**Comment by stropitek** [comment]
Save LGTM

Loading the `.ium` is not yet implemented

**Comment by lpatiny** [comment]
@stropitek Could we merge this PR ? Loading the .ium should be part of another PR.

**Pull request merged**